packit_service/worker/build/build_helper.py

```diff
@@ -373,9 +417,30 @@ def _report(
         The status reporting should be done through this method
         so we can extend it in subclasses easily.
         """
-        self.status_reporter.report(
-            description=description, state=state, url=url, check_names=check_names,
-        )
+        if isinstance(self.project, GitlabProject) and not self.is_reporting_allowed:
+            # "Packit-User" does not have access to project.
+            self.request_project_access()
+            self.report_status_by_comment(state, url, check_names)
+        else:
+            self.status_reporter.report(
+                description=description, state=state, url=url, check_names=check_names,
+            )
+
+    def report_status_by_comment(
+        self, state: CommitStatus, url: str, check_names: Union[str, list, None],
+    ) -> None:
+        """
+        Reporting build status with MR comment if no permission to the fork project
+        """
+        build_states = [CommitStatus.error, CommitStatus.success, CommitStatus.failure]
+
+        if isinstance(check_names, str):
```

Comment:
What about the `check_names is None` scenario. (At least from the type annotation, it looks like we can get that.)

Reply:
Ah, I have changed the type annotation.

Reply:
Solved it here - [R438-R439](https://github.com/packit/packit-service/pull/740/files#diff-599fa650469aeda8bea187fa74ed9502R438-R439)

Reply:
If I read that correctly, we don't want to process status that is final (`error`/`success`/`failure`).

- What will be done for those "final" states?
- Is it possible to move this logic (= `if state not in ...` condition) out of this method? Is it too complicated?
- Can we add a comment and/or update docstring to say why we are doing this?

Reply:
@lachmanfrantisek 
We only want to report status by comment if it's a final state (error / success / failure).

```
if state not in build_states or check_names is None:
    return
```
The above condition was still valid, as we are not doing anything if `state` has any value other then (error/success/failure).

Anyways I have updated and moved that part as suggested. 

I think this might be easier to read -
```
if state in build_states and check_names is not None:
    self.status_reporter.report_status_by_comment(state, url, check_names)
```

Reply:
Ah, yes. My bad...;)

Thanks!